frb/figures/dm.py

```diff
@@ -0,0 +1,208 @@
+
+from ne2001 import density
+
+def sub_cartoon(ax1, ax2, coord, zFRB, halos=False, host_DM=50., ymax=None,
+                IGM_only=True,
+                M31=False, fg_halos=None, dsmx=0.05, FRB_DM=None, yscl = 0.97):
+    """
+    Cartoon of DM cumulative
+
+    Plot of increasing DM from Earth to the FRB
+
+    Args:
+        ax1 (matplotlib.Axis):
+            First axis.  Used for Milky Way and local group
+        ax2 (matplotlib.Axis):
+            Second axis.  Used for Cosmic and Host
+        coord (astropy.coord.SkyCoord):
+            Coordinate of the FRB used with ne2001
+        zFRB (float):
+            Redshift of the FRB
+        halos (?, optional):
+            Not implemented!
+        host_DM (float):
+            DM to use for the Host
+        ymax (tuple or list):
+            ymin, ymax values for the y-axis
+        IGM_only (bool, optional):
+            Use only the IGM for DM_Cosmic, i.e. ignore the presumed average halo contribution
+        M31 (bool, optional):
+            Include M31 in the calculation?
+            NOT IMPLEMENTED RIGHT NOW
+        fg_halos (dict or Table):
+            Used to add to DM_IGM
+            Keys must include 'z' 'DM' 'lbl'
+        dsmx (float): Padding on the x-axis;  Gpc
+        FRB_DM (float): Observed value;  sets ymax = FRB_DM+50
+    """
+    if halos:
+        embed()
+
+    gcoord = coord.transform_to('galactic')
+    l, b = gcoord.l.value, gcoord.b.value
+
+    ds = []  # kpc
+    DM_cumul = []
+
+    # ISM
+    ne = density.ElectronDensity()  # **PARAMS)
+    for ss in np.linspace(2., 4, 5):  # log pc
+        idd = 10. ** ss / 1e3  # kpc
+        iDM = ne.DM(l, b, idd)
+        # Append
+        ds.append(idd)  # kpc
+        DM_cumul.append(iDM.value)
+        # print(idd, iDM)
+    max_ISM = DM_cumul[-1]
+
+    # MW
+    Mhalo = np.log10(1.5e12)  # Boylan-Kolchin et al. 2013
+    f_hot = 0.75  # Allows for disk + ISM
+    c = 7.7
+    mnfw_2 = ModifiedNFW(log_Mhalo=Mhalo, f_hot=f_hot, y0=2, alpha=2, c=c)
+    # Zero out inner 10kpc
+    mnfw_2.zero_inner_ne = 10.  # kpc
+    params = dict(F=1., e_density=1.)
+    model_ne = density.NEobject(mnfw_2.ne, **params)
+    for ss in np.linspace(1., np.log10(mnfw_2.r200.value), 5):  # log kpc
+        idd = 10. ** ss  # kpc
+        iDM = model_ne.DM(l, b, idd).value
+        # Add it in
+        if idd == ds[-1]:
+            DM_cumul[-1] = DM_cumul[-1] + iDM
+        else:
+            ds.append(idd)
+            DM_cumul.append(max_ISM + iDM)
+
+    DM_ISM_Halo = DM_cumul[-1]
+
+    '''
+    # M31
+    m31 = M31()
+    a,c =1,0
+    x0, y0 = m31.distance.to('kpc').value, 0. # kpc (Riess, A.G., Fliri, J., & Valls - Gabaud, D. 2012, ApJ, 745, 156)
+    sep = m31.coord.separation(coord)
+    atan = np.arctan(sep.radian)
+    b = -1 * a / atan
+    M31_Rperp = np.abs(a * x0 + b * y0 + c) / np.sqrt(a ** 2 + b ** 2)  # kpc
+    zval, M31_DM_cumul = m31.Ne_Rperp(M31_Rperp * u.kpc, rmax=1., cumul=True)
```

Comment:
remove this commented out code?

Reply:
kept the code and wrapped it with a `NotImplemented` error.